lambda-api/src/main/java/com/formkiq/stacks/api/handler/DocumentsCompressRequestHandler.java

```diff
@@ -0,0 +1,112 @@
+import com.formkiq.aws.services.lambda.ApiGatewayRequestEvent;
+import com.formkiq.aws.services.lambda.ApiGatewayRequestEventUtil;
+import com.formkiq.aws.services.lambda.ApiRequestHandlerResponse;
+import com.formkiq.aws.services.lambda.ApiMapResponse;
+import com.formkiq.module.lambdaservices.AwsServiceCache;
+
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Map;
+import java.util.UUID;
+
+import static com.formkiq.aws.services.lambda.ApiResponseStatus.SC_CREATED;
+import static java.util.Map.entry;
+
+/** {@link ApiGatewayRequestHandler} for "/documents/compress". */
+public class DocumentsCompressRequestHandler
+    implements ApiGatewayRequestHandler, ApiGatewayRequestEventUtil {
+
+  @Override
+  public String getRequestUrl() {
+    return "/documents/compress";
+  }
+
+  @Override
+  public ApiRequestHandlerResponse post(final LambdaLogger logger,
+      final ApiGatewayRequestEvent event, final ApiAuthorization authorization,
+      final AwsServiceCache awsServices) throws Exception {
+    DynamicObject requestBodyObject = fromBodyToDynamicObject(event);
+
+    final String siteId = authorization.siteId();
+    final String compressionId = UUID.randomUUID().toString();
+    final String compressionTaskS3Key = getS3Key(siteId, compressionId, false);
+    S3Service s3 = awsServices.getExtension(S3Service.class);
+    final String stagingBucket = awsServices.environment("STAGE_DOCUMENTS_S3_BUCKET");
+    final String downloadUrl =
+        getArchiveDownloadUrl(s3, stagingBucket, getS3Key(siteId, compressionId, true));
+    final DynamicObject taskObject =
+        getS3TaskObject(requestBodyObject, siteId, compressionId, downloadUrl);
+
+    putObjectToStaging(s3, stagingBucket, compressionTaskS3Key, GSON.toJson(taskObject));
+    ApiMapResponse response =
+        new ApiMapResponse(Map.of("compressionId", compressionId, "downloadUrl", downloadUrl));
+    return new ApiRequestHandlerResponse(SC_CREATED, response);
+  }
+
+  private String getS3Key(final String siteId, final String compressionId, final boolean isZip) {
+    final String tempPrefix = String.format("tempfiles/%s", siteId == null ? "" : siteId + "/");
+    final String fileType = isZip ? ".zip" : ".json";
+    return tempPrefix + compressionId + fileType;
+  }
+
+  private DynamicObject getS3TaskObject(final DynamicObject requestBodyObject, final String siteId,
+      final String compressionId, final String downloadUrl) {
+    final String documentIdsKey = "documentIds";
+    final Object documentIds = requestBodyObject.get(documentIdsKey);
+    final String compressionIdKey = "compressionId";
+    final String downloadUrlKey = "downloadUrl";
+    final String siteIdKey = "siteId";
+    return new DynamicObject(Map.ofEntries(entry(documentIdsKey, documentIds),
+        entry(compressionIdKey, compressionId), entry(downloadUrlKey, downloadUrl),
+        entry(siteIdKey, siteId == null ? "null" : "siteId")));
```

Comment:
default SiteId is SiteIdGenerator.DEFAULT_SITE_ID

Reply:
Done.